index.bs

```diff
@@ -284,30 +281,28 @@ The <dfn method for=IdleDetector>requestPermission()</dfn> method steps are:
 
 The <dfn method for=IdleDetector>start(|options|)</dfn> method steps are:
 
-1.  Let |result| be [=a new promise=].
 1.  If the [=relevant global object=]'s [=associated Document=] is not [=allowed
-    to use=] the [=policy-controlled feature=] named `"idle-detection"`
-    [=reject=] |result| with a "{{NotAllowedError}}" {{DOMException}} and return
-    |result|.
-
+    to use=] `"idle-detection"`, return [=a promise rejected with=] a
+    "{{NotAllowedError}}" {{DOMException}}.
     <wpt>
       idle-detection-allowed-by-feature-policy-attribute-redirect-on-load.https.sub.html
       idle-detection-allowed-by-feature-policy-attribute.https.sub.html
       idle-detection-allowed-by-feature-policy.https.sub.html
       idle-detection-default-feature-policy.https.sub.html
       idle-detection-disabled-by-feature-policy.https.sub.html
     </wpt>
-1.  If |this|.{{IdleDetector/[[state]]}} is not `"stopped"`, [=reject=] |result|
-    with an "{{InvalidStateError}}" {{DOMException}} and return |result|.
+1.  If |this|.{{IdleDetector/[[state]]}} is not `"stopped"`,
+    return [=a promise rejected with=] an "{{InvalidStateError}}" {{DOMException}}.
     <wpt>
       interceptor.https.html
     </wpt>
 1.  Set |this|.{{IdleDetector/[[state]]}} to `"starting"`.
-1.  If |options|[{{"threshold"}}] is less than 60,000 [=reject=] |result| with
-    {{TypeError}} and return |result|.
+1.  If |options|[{{"threshold"}}] is less than 60,000, return [=a promise rejected with=]
```

Comment:
Unrelated... but shouldn't this check happen before setting `IdleDetector.[[state]` to `"starting"`?

Reply:
Good catch, checking whether `options['signal']` is aborted should also happen before changing the state to `"starting"`.

Reply:
filed https://github.com/WICG/idle-detection/issues/47 so we don't forget as a follow up.